ast/policy.go

```diff
@@ -1000,16 +1017,26 @@ func (head *Head) Equal(other *Head) bool {
 }
 
 func (head *Head) String() string {
+	return head.stringWithOpts(toStringOpts{})
+}
+
+func (head *Head) stringWithOpts(opts toStringOpts) string {
 	buf := strings.Builder{}
 	buf.WriteString(head.Ref().String())
 
 	switch {
 	case len(head.Args) != 0:
 		buf.WriteString(head.Args.String())
 	case len(head.Reference) == 1 && head.Key != nil:
-		buf.WriteRune('[')
-		buf.WriteString(head.Key.String())
-		buf.WriteRune(']')
+		switch opts.regoVersion {
+		case RegoV0:
+			buf.WriteRune('[')
+			buf.WriteString(head.Key.String())
+			buf.WriteRune(']')
+		default:
+			buf.WriteString(" contains ")
+			buf.WriteString(head.Key.String())
```

Comment:
This "contains" and the "contains" in line 1022/1048 cannot overlap, I suppose? 🤔

Reply:
Not through parsing, I think. But since a developer can build their own rule head programmatically, I'll add in a check just to make sure.